### PR TITLE
Expose information on whether transaction relayed is enabled in getnetwork

### PR DESCRIPTION
**Pull request by ghost**
adding CNode::fRelayTxes to getnetworkinfo under the name of localrelay (bool)

updated pull #7839
#7771

### CODE REVIEW

**Comment by dragongem45** [comment]
a new fix has been applyed to this code

**Comment by sipa** [comment]
@dragongem45 If you find more things to fix in the future, you can just force push to the github branch associated with it, you don't need to close and open a new pull request.

**Comment by dragongem45** [comment]
is there a way to restart travis-ci biulds

**Comment by sipa** [comment]
@dragongem45 Yes, by pushing a new commit.

**Comment by dragongem45** [comment]
do you have any ideas on what i am doing wrong

**Comment by sipa** [comment]
See https://travis-ci.org/bitcoin/bitcoin/jobs/121714563

```
In file included from main.h:16:0,
                 from checkpoints.cpp:9:
net.h: In static member function ‘static bool CNode::GetfRelayTxes()’:
net.h:360:10: error: invalid use of member ‘CNode::fRelayTxes’ in static member function
net.h:362:16: error: from this location
```

Can you try compiling your own code before submitting a patch?

**Comment by dragongem45** [comment]
ok

**Comment by MarcoFalke** [comment]
@dragongem45 Please build and test your changes locally. Also make sure to squash the commits using git locally. If there are any issues we can help you in the IRC.

**Comment by dragongem45** [comment]
sorry its a part of my IDE i'm going to disable it

**Comment by dragongem45** [comment]
compile worked. now there is a python test error

**Comment by sipa** [comment]
It seems you're turning getnetworkinfo into something that returns information about all peers. That's not correct. fRelayTxs is a per-peer variable, so if you want to expose it, it will go in getpeerinfo which lists information over all peers.

**Comment by MarcoFalke** [comment]
@dragongem45 Please squash your commits:

```sh
git clone https://github.com/dragongem45/bitcoinclassic.git bitcoin
cd bitcoin
git fetch --all
git checkout patch-2
git rebase --interactive ea4c315~
[Select "f" for all lines but the first]
git push origin patch-2 -f

**Comment by sipa** [comment]
The `relaytxes` field in getnetworkinfo now reports whether we have at least one peer which has the fRelayTxs flag set. Wouldn't it be more logical to put it in getpeerinfo, where it gets reported for every peer individually?

**Comment by laanwj** [comment]
> The relaytxes field in getnetworkinfo now reports whether we have at least one peer which has the fRelayTxs flag set. Wouldn't it be more logical to put it in getpeerinfo, where it gets reported for every peer individually?

I think it's not implemented correctly. The original idea behind exposing this on the interface is so that applications know whether *we* relay transactions. So what value of relaytxes we send in our version message.

Per-peer information could be useful too, but that's orthogonal.

File: src/net.h [hunk 1]
```diff
@@ -769,8 +769,6 @@ class CNode
     static uint64_t GetMaxOutboundTimeLeftInCycle();
 };
 
-
-
```
Comment (MarcoFalke) [inline src/net.h]: @dragongem45 Are you still working on this?

Please remove this diff and squash your commits.

**Comment by sipa** [comment]
Agree with @laanwj

**Comment by laanwj** [comment]
Closing in favor of #8049

**Pull request closed without merging**